### PR TITLE
add reboot option to vm state command

### PR DESCRIPTION
**Pull request by mheidenr**
All errors that can occur are documented in https://www.vmware.com/support/developer/converter-sdk/conv50_apireference/vim.VirtualMachine.html#rebootGuest

### CODE REVIEW

File: lib/chef/knife/vsphere_vm_state.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@ class Chef::Knife::VsphereVmState < Chef::Knife::BaseVsphereCommand
   option :state,
          short: '-s STATE',
          long: '--state STATE',
-         description: 'The power state to transition the VM into; one of on|off|suspended'
+         description: 'The power state to transition the VM into; one of on|off|suspended|restart'
```
Comment (swalberg) [inline lib/chef/knife/vsphere_vm_state.rb]: Should this be `reboot`?

File: lib/chef/knife/vsphere_vm_state.rb [hunk 2]
```diff
@@ -110,6 +110,9 @@ def run
       when 'reset'
         vm.ResetVM_Task.wait_for_completion
         puts "Reset virtual machine #{vmname}"
+      when 'reboot'
+        vm.RebootGuest
```
Comment (swalberg) [inline lib/chef/knife/vsphere_vm_state.rb]: The other options in this command all wait for the command to complete in some manner. Is it advantageous not to do it here?

File: lib/chef/knife/vsphere_vm_state.rb (hunk 1 above)
Comment (mheidenr) [inline lib/chef/knife/vsphere_vm_state.rb]: yes, of course. I'll fix it

File: lib/chef/knife/vsphere_vm_state.rb (hunk 2 above)
Comment (mheidenr) [inline lib/chef/knife/vsphere_vm_state.rb]: Unfortunately, this method doesn' t return a task object.

From [API documentation](https://www.vmware.com/support/developer/converter-sdk/conv50_apireference/vim.VirtualMachine.html#rebootGuest):
>RebootGuest
>Issues a command to the guest operating system asking it to perform a reboot. Returns immediately and does not wait for the guest operating system to complete the operation.

I tried to implement a wait option, but I didn't have any idea how to achieve this - I think VMware didn't as well.

Comment (swalberg) [inline lib/chef/knife/vsphere_vm_state.rb]: Aha, thanks for that explanation. Also, thanks for putting this PR together! Could I trouble you to squash it down to one commit and I'll merge it?

Thanks.

**Comment by mheidenr** [comment]
@swalberg done, thanks for your effort

**Pull request merged**